flow/core/kernel/kernel.py

```diff
@@ -0,0 +1,94 @@
+"""Script containing the Flow kernel object for interacting with simulators."""
+
+
+class Kernel(object):
+    """Kernel for abstract function calling across traffic simulator APIs.
+
+    The kernel contains four different subclasses for distinguishing between
+    the various components of a traffic simulator.
+
+    * simulation: controls starting, loading, saving, advancing, and resetting
+      a simulation in Flow (see flow/core/kernel/simulation/base.py)
+    * scenario: stores network-specific information (see
+      flow/core/kernel/scenario/base.py)
+    * vehicle: stores and regularly updates vehicle-specific information. At
+      times, this class is optimized to efficiently collect information from
+      the simulator (see flow/core/kernel/vehicle/base.py).
+    * traffic_light: stores and regularly updates traffic light-specific
+      information (see flow/core/kernel/traffic_light/base.py).
+
+    The above kernel subclasses are designed specifically to support
+    simulator-agnostic state information calling. For example, if you would
+    like to collect the vehicle speed of a specific vehicle, then simply type:
+
+    >>> k = Kernel(simulator="...")  # a kernel for some simulator type
+    >>> veh_id = "..."  # some vehicle ID
+    >>> speed = k.vehicle.get_speed(veh_id)
+
+    In addition, these subclasses support sending commands to the simulator via
+    its API. For example, in order to assign a specific vehicle a target
+    acceleration, type:
+
+    >>> k = Kernel(simulator="...")  # a kernel for some simulator type
+    >>> veh_id = "..."  # some vehicle ID
+    >>> k.vehicle.apply_acceleration(veh_id)
```

Comment:
apply_acceleration is moving to the vehicle class? Which class subclasses GymEnv?

Reply:
Also, not sure if I agree with the variable name k for the kernel. While it's abundantly clear to us what it means, it may add additional confusion for new developers.

Reply:
So the gym env will still be in base env, with all the relevant calls available from there. What this is meant to do is when you pass a command to the simulator, the command isn't sent to a method within the gym class, but rather to a method within the correct simulation kernel. That way `apply_acceleration` can work for any simulator without us having to modify the envs.

Reply:
In terms of k, we'll have documentation on this, but if you'd rather we call it something else, e.g. `self.kernel`, I'd be down too. It just makes calling commands longer